File: 202 Happy Number.java
class Solution {
    public boolean isHappy(int n) {
        Set<Integer> set = new HashSet<>();
        int next;
        while (!set.contains(n)) {
            if (n == 1) return true;
            set.add(n);
            next = 0;
            while (n != 0) {
                int remain = n % 10;
                n /= 10;                
                next += Math.pow(remain, 2);
            }
            n = next;
        }
        return false;
    }
}

File: 202 Happy Number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        loop = set()
        while n not in loop:
            loop.add(n)
            sqSum = 0
            while n > 0:
                remain = n % 10
                sqSum += remain ** 2
                n //= 10
            if sqSum == 1:
                return True
            n = sqSum
        return False